src/preprocessing/passes/non_clausal_simp.cpp

```diff
@@ -57,68 +67,44 @@ PreprocessingPassResult NonClausalSimp::applyInternal(
   Assert(!options::unsatCores() && !options::fewerPreprocessingHoles());
 
   d_preprocContext->spendResource(options::preprocessStep());
+  unsigned substs_index = d_preprocContext->getSubstitutionsIndex();
+  bool noconflict;
+  std::vector<Node> learned_literals;
 
-  theory::booleans::CircuitPropagator* propagator =
-      d_preprocContext->getCircuitPropagator();
-
-  for (size_t i = 0, size = assertionsToPreprocess->size(); i < size; ++i)
-  {
-    Trace("non-clausal-simplify") << "Assertion #" << i << " : "
-                                  << (*assertionsToPreprocess)[i] << std::endl;
+  if (options::skeletonPreprocessing()){
+    // solve the boolean skeleton using cryptominisat
+    std::tie(noconflict, learned_literals) =
+        preprocessByCryptoMinisat(assertionsToPreprocess, substs_index);
   }
-
-  if (propagator->getNeedsFinish())
+  else
   {
-    propagator->finish();
-    propagator->setNeedsFinish(false);
+    // default
+    // solve the boolean skeleton using circuit propagator
+    std::tie(noconflict, learned_literals) =
+        preprocessByCircuitPropagator(assertionsToPreprocess, substs_index);
   }
-  propagator->initialize();
 
-  // Assert all the assertions to the propagator
-  Trace("non-clausal-simplify") << "asserting to propagator" << std::endl;
-  unsigned substs_index = d_preprocContext->getSubstitutionsIndex();
-  for (size_t i = 0, size = assertionsToPreprocess->size(); i < size; ++i)
+  if (!noconflict)
```

Comment:
How about renaming `noconflict` to `ok`?

Reply:
sounds good!